File: backend/app/models/meal_model.py
from pydantic import BaseModel, HttpUrl
from typing import Optional, List


class Meal(BaseModel):
    id: Optional[str]
    canteen_id: str
    name: str
    category: str
    price: float
    calories: Optional[int]
    ingredients: Optional[List[str]] = []
    diet_type: str  # "veg" | "non-veg" | "vegan"
    image_url: Optional[HttpUrl]
    is_available: bool = True

File: khaikhai/lib/models/meal_model.dart
class Meal {
  final String id;
  final String name;
  final String category;
  final double price;
  final int calories;
  final String? imageUrl;

  Meal({
    required this.id,
    required this.name,
    required this.category,
    required this.price,
    required this.calories,
    this.imageUrl,
  });

  factory Meal.fromJson(Map<String, dynamic> json) {
    return Meal(
      id: json['_id'],
      name: json['name'],
      category: json['category'],
      price: (json['price'] as num).toDouble(),
      calories: json['calories'] ?? 0,
      imageUrl: json['image_url'],
    );
  }
}
